### DISCUSSION TITLE
Source Distributions for fastf1 >= 3.0.6

### DISCUSSION THREAD

**Discussion by daylinmorgan** [theOehrly/Fast-F1#411] [General]
When updating the `conda-forge` package for `fastf1` to v3.0.6 I ran into an issue since there doesn't seem to be a [source distribution](https://pypi.org/project/fastf1/3.0.6/#files).

Though the `release.yml` workflow does include an `sdist`. Just want to confirm if this was a manually uploaded release and that future versions will have a source distribution and the `ci` didn't fail to upload it in this case.

**Comment by theOehrly** [comment]
Yes, it usually includes a source distribution. But somehow the release workflow decided not to trigger. Don't ask me why. It simply didn't even start. (It did trigger less than 1 hour earlier for v3.0.5 and there were no changes to the CI after that.)

I decided to just upload manually from my local machine. But there, the source distribution include most of the development environment as well. So in the end I just deleted the source distribution from PyPi, because I really needed to go to bed 😅 Sorry for causing any problems by doing that.

**Comment by daylinmorgan** [comment]
No worries. Seems like github has had some intermittent issues with actions lately. Just wanted to make sure future versions will still have source distributions for down-stream packaging (like on `conda-forge`).

**Reply by theOehrly** [reply]
I didn't know if anyone actually depends on these. But next time, I will try not to make a new release in the middle of the night. So that if anything goes wrong again, I can fix it properly 😅

**Reply by daylinmorgan** [reply]
Ya it's nice to include for pure python packages. 

I typically tie PYPI releases to version tags and not releases. Not sure if that would make a difference in the future for you.

```diff
on:
-  release:
-    types: [released]
+  push:
+   tags:
+     - "v*.*.*"

```

**Reply by theOehrly** [reply]
Is there any practical advantage for this? I'm planning to use the Github release feature anyway, so I don't think this would make any difference?

**Comment by daylinmorgan** [comment]
No they should work the same. You might have just created the release during one of their recent server issues.
